IdentityCore/src/MSIDError.h

```diff
@@ -39,5 +40,5 @@ typedef NS_ENUM(NSInteger, MSIDErrorCode)
     removeMeWhenThingsAreAdded = -10000,
 };
 
-extern NSError *MSIDCreateError(NSString *domain, NSInteger code, NSString *errorDescription, NSString *oauthError, NSString *subError, NSError *underlyingError);
+extern NSError *MSIDCreateError(NSString *domain, NSInteger code, NSString *errorDescription, NSString *oauthError, NSString *subError, NSError *underlyingError, NSDictionary *httpHeaders);
```

Comment:
I'd like it to be more generic: e.g. other than passing httpHeaders directly, passing extraUserInfo. Http headers will need to be passed only in one place, but passing additional userInfo can be reused in multiple situations. 

e.g. in MSAL we added:

`extern NSError *MSALCreateError(NSString *domain, NSInteger code, NSString *errorDescription, NSString *oauthError, NSString *subError, NSError* underlyingError, NSDictionary* additionalUserInfo);`

Reply:
I was thinking the same when I did the coding :+1: